### ISSUE TITLE
Extension Proposal: Sub-properties of Identifier

### ISSUE THREAD

**Issue by fili** [schemaorg/schemaorg#2129]
My 2 cents:

As part of Brand we can add an identifier. Identifiers have several interesting sub-properties such as taxID, productID, duns, leiCode, serialNumber, etc. What I am missing is "Trademark". 

Often a Brand is also registered as a Trademark, so it make sense if the Brand can be annotated as legally registered trademark - this would be similar to how an organisation is annotated as legally registered with leiCode or taxID. 

What I expect to be able to annotate is at minimum the **trademarkName**, **trademarkID** and **trademarkOfficeCode** (often a two letter code). Additionally countrycodes for _designated territory_, _classes_ (based on Nice classes) and _trademarkType_: if the trademark is a graphic representation, then a URL to the image may be useful. 

What do you think?

**Comment by MichaelAndrews-RM** [comment]
@fili Can you provide more background on how you'd like to use these properties?  

To me, all brands mentioned using the vocabulary are likely be trademarked.  The most important linkage is to  connect the brand to an organization.  There are many brand names that are used by more than one corporation.  Brand names don't need to be unique, as long as they don't mislead customers.  Giant brand bicycles are unlikely to be confused with Giant brand plastic bags.  Often common brand names are owned by different parties in different countries. These relationships can be disambiguated by linking the brand to the organization.  The same product can also be sold under different names in different markets -- a common practice in pharmaceuticals.  

When one gets down to the level of trademarks, there's a massive amount of detail involved, since each country awards trademarks independently and may impose limits on the scope of protections offered.  One country may allow a brand name to be used for any product, while another will allow it to be used for only certain products.   It would not seem desirable to re-create a database of all trademarks of a corporation for all countries of the world by noting each fact using the schema.org vocabulary.  Is it possible to cite one source for all the trademarks associated with a brand?  Is there any linked data registry of trademarks?

**Comment by fili** [comment]
@MichaelAndrews-RM Thanks for your reply. I will try to address some of your questions.

> To me, all brands mentioned using the vocabulary are likely be trademarked. 

That is exactly I want to clarify. "Likely" is not guaranteed. It is an assumption I like to see supported by fact.

> Brand names don't need to be unique, as long as they don't mislead customers. Giant brand bicycles are unlikely to be confused with Giant brand plastic bags.

Two thoughts here: 
1. True, brand names do not need to be unique. But trademark annotation would not be a required field, so this proposal should not be in the way of adding multiple brand names, including those without trademark registration. This proposal is about adding the option to add verifiable data, like a taxID.
2. This is also where (Nice) classes come in, and differentiate Giant brand bicycles from Giant brand plastic bags because of the different classes. 

> Often common brand names are owned by different parties in different countries.

This is where the trademarkOfficeCode comes in.

> These relationships can be disambiguated by linking the brand to the organization.

Except there is no verification if the brand name is indeed belonging to the organization. Adding the trademark information would allow for independent verification checks whereas now it is just a unfounded claim. Any organization can claim the brand name Giant brand plastic bags. However only one organization can own a trademark in the trademark registry of a trademarkOfficeCode, so this proposal would make the claim potentially verifiable (something it is not currently).

> The same product can also be sold under different names in different markets -- a common practice in pharmaceuticals.

Also here it is where trademarkOfficeCode comes in. Actually this is for me an argument in favor of trademark information, to clarify which registered brands are registered where. Your comment already suggests that an organization would add multiple brand names in this case, so now we can verify it in the different markets. 

> When one gets down to the level of trademarks, there's a massive amount of detail involved, since each country awards trademarks independently and may impose limits on the scope of protections offered. One country may allow a brand name to be used for any product, while another will allow it to be used for only certain products.

True, but this is again why I think trademarkOfficeCode is a necessary field. Schema does not need to write down all the requirements for registration. It is not doing this for taxID as different countries have different regulations on how to get a taxID, yet this is fine to add as an identifier - same would apply to this proposal. Instead this proposal would give the minimum required information to be able to confirm the trademark by stating trademarkName (what - this can possibly be replaced by brandName), trademarkID (unique ID) and trademarkOfficeCode (where).

> It would not seem desirable to re-create a database of all trademarks of a corporation for all countries of the world by noting each fact using the schema.org vocabulary.

I think if something is registered as a brand, and we mark that up as a brand name, it would be good to support this by fact if the brand is registered as a trademark. Trademarks have been around for quite some time, and I am frankly surprised it is not included in Schema, especially as taxID or leiCode is.

> Is it possible to cite one source for all the trademarks associated with a brand? Is there any linked data registry of trademarks?

Good questions. There are several different search databases. Including those supported by the [WIPO](https://www.wipo.int/amc/en/trademark/).

**Comment by fili** [comment]
> Actually, the most important is to show "ownership" of a brand with Date qualifiers (Brands can change ownership through acquisitions for instance). We had this on Freebase, and it's also present in Wikidata.

Although you raise a good point, even historic trademark information can be useful. As trademarks generally last at least 10 years, I think this is less of a problem right now. But yes, we can add date fields as optional fields. I think however the current status can be checked if we have the necessary trademarkID and trademarkOfficeCode fields, as the ID is unique and the office code should make programmatic lookups possible.

**Comment by MichaelAndrews-RM** [comment]
> Actually, the most important is to show "ownership" of a brand with Date qualifiers (Brands can change ownership through acquisitions for instance).

We potentially have two different concepts at play here: trademark "ownership" and trademark "registration".  

In one case, a corporation asserts they own a trademark, and may sell it privately to another, and this may not involve any government organization.  In the US, some trade names are "trademarks" (TM) which have common law status, but they are not registered (R) with the United States Patent and Trademark Office (USPTO).  In the other case, a registered mark implies the USPTO (in the US case) has accepted the claim.  I don't know the situation in other countries.  It seems likely that different government bodies around the world will reach different conclusions about whether a trademark qualifies for registration.  So is ownership (however claimed) more important to indicate, or registration?  

Some brands that do not sell products but sell services use a "service mark" (SM).

**Comment by fili** [comment]
The focus of the proposal is not ownership. Instead the proposal is about marking a brand up with an identifiable and verifiable registration ID, hence this proposal is to add this as a sub-property to identifier, not to owns. 

However it is true that one can lead to another and ownership is something we can extend to in the future.

**Comment by thadguidry** [comment]
@fili So ... as a fun example, you are asking for a property to hold something like the US Serial Number ["86400668"](http://tsdr.uspto.gov/#caseNumber=86400668&caseType=SERIAL_NO&searchType=statusSearch) on the Brand "KUBERNETES" currently owned by The Linux Foundation (previously owned by Google) ?

or ["73725688"](http://tsdr.uspto.gov/#caseNumber=73725688&caseType=SERIAL_NO&searchType=statusSearch) for your example of "Giant" brand bicycle company ...

and those numbers would be held in a proposed property called "trademarkID" ?

**Comment by thadguidry** [comment]
Incidentally, time series (date mediation) is even a factor with USPTO serial numbers...

> serial number
> 
> a number assigned to a patent application when it is filed. A serial number is usually used together with a two digit series code to distinguish between applications filed at different times.
>
> series code
> 
> a two digit code representing a period of time. Application Filing Date - examples: 01/01/1979-12/31/1986 Series 06; 01/01/1987-12/31/1992 Series 07;01/01/1993-12/31/1997 Series 08; 01/01/1998-November 2001 Series 09; December 2001-Present Series 10

**Comment by fili** [comment]
@thadguidry No, because those numbers are not the registration numbers but rather the serial numbers. For KUBERNETES the registration number is [4754865](https://www.tmdn.org/tmview/get-detail?st13=US500000086400668) and for GIANT it is 1634758. It would look like something like this:

```
{ 
  "@context" : "http://schema.org",
  "@type" : "Organization",
  "name" : "THE LINUX FOUNDATION",
  "brand": {
      "@type": "Brand",
      "name": "Kubernetes",
      "identifier": {
          "@type": "Trademark"
          "trademarkID": "4754865",
          "trademarkOfficeCode": "USPTO",
          "class": "9",
          "applicationNumber": "86400668"	
        }
  }

  [...]

}
```
Does this make sense?

**Comment by fili** [comment]
@thadguidry Another good point. So we should rename trademarkID to registrationNumber and allow for both applicationNumber and/or registrationNumber. Check out this one for [Kubernetes](https://euipo.europa.eu/eSearch/#details/trademarks/013794904).

For example:

```
{ 
  "@context" : "http://schema.org",
  "@type" : "Organization",
  "name" : "THE LINUX FOUNDATION",
  "brand": {
      "@type": "Brand",
      "name": "Kubernetes",
      "identifier": [{
          "@type": "Trademark"
          "registrationNumber": "4754865",
          "applicationNumber": "86400668",
          "trademarkOfficeCode": "USPTO",
          "class": "9"
        }, {
          "@type": "Trademark"
          "registrationNumber": "013794904",
          "trademarkOfficeCode": "EUTM",
          "class": "9"
        }]
  }

  [...]

}
```

**Comment by fili** [comment]
@thadguidry BTW I am not sure if the registration number of the Agent/Attorney you mentioned does not refer to the docket number under the Correspondence details. The reason why I am wondering is because [the example you referred to](http://tsdr.uspto.gov/#caseNumber=87512521&caseSearchType=US_APPLICATION&caseType=SERIAL_NO&searchType=documentSearch) is not registered but still in process of being registered. When a trademark gets registered it seems to get a registration number assigned in the registration certificate [example](http://tsdr.uspto.gov/documentviewer?caseId=sn86400668&docId=ORC20150616010418#docIndex=0&page=1)

As I understand it:

> A US trademark is assigned a "Serial Number" upon filing the application. Once registered, it also gets a "Registration Number". A "trademark number" probably refers to the serial number, since all trademarks, including unregistered ones, will have that as soon as they are filed

So the registration number may be better focus for identifying a registered trademark.

Do you have more examples? Including non-US focused?

**Comment by thadguidry** [comment]
@fili Sorry, I thought you were an expert in this field.  I am not either, but have learned much over the last 10 years with my own previous research and speaking to a Patent Attorney on more than one occasion, which does not make me an expert, in the USA not even Patent Attorneys are experts, but instead only USPTO staff. :-)  Anyways...

In nearly all systems... a "Mark" is given a unique ID.  In the USPTO system, a "Mark" is requested to be federally registered by submitting an application...in doing so, it is 1st given a US Serial Number that uniquely identifies the APPLICATION of the "Mark" to be registered AND all documents related to it throughout its entire lifespan .... as @MichaelAndrews-RM told you, it could be a Servicemark, Trademark, etc.

It will completely look like this...missing only the Registration Number...
http://tsdr.uspto.gov/documentviewer?caseId=sn86400668&docId=ORC20150616010418#docIndex=4&page=1

After it is registered, it will get a Registration Number and certificate...
http://tsdr.uspto.gov/documentviewer?caseId=sn86400668&docId=ORC20150616010418#docIndex=0&page=1

**Application Serial Number** (commonly just called Serial Number within the USPTO, but they also call this their Case Id, if you didn't know)
**Registration Number** (the registration number assigned after USPTO grants registration into the federal system that the "Mark" is registered to the Registrant / Owner. )
**Registration Number** (the registration number for a Patent Attorney that is registered with the USPTO
OED office.  https://oedci.uspto.gov/OEDCI/ and what I was referring to previously, so in a TSDR search, you WILL see the same Registration Number for multiple applications/marks, because it is the same Patent Attorney that is filing these quite often for companies)

The use of the term Registration Number is used in multiple systems of the USPTO (yes, very confusing, I know) ... its best to use a full name like "markRegistrationID" for the registered "Mark" ID to avoid confusion, or just simply "markID" and a good definition to go into Schema.org

I'm only familiar with US application process.

**Comment by MichaelAndrews-RM** [comment]
So far we have only discussed how the name of a Brand is a trademark.  But [WIPO](https://www.wipo.int/trademarks/en/) states:

>A word or a combination of words, letters, and numerals can perfectly constitute a trademark. But trademarks may also consist of drawings, symbols, three-dimensional features such as the shape and packaging of goods, non-visible signs such as sounds or fragrances, or color shades used as distinguishing features – the possibilities are almost limitless.

For example, perhaps a triangular shaped chocolate is trademarked.  How would that be represented?  And how could humans or machines even query shape characteristics to get an answer about their trademark status?

**Comment by fili** [comment]
@MichaelAndrews-RM You raise a good question:

> For example, perhaps a triangular shaped chocolate is trademarked. How would that be represented? And how could humans or machines even query shape characteristics to get an answer about their trademark status?

However I see this as outside the scope of this proposal. Right now this proposal is about adding an type of identifier to a Brand. What the brand is, or how it is described, depends on the Brand schema. I assume in case of graphics, as I mentioned earlier, you can use the URL to the graphic. Also graphics often, although not always, have a written description which is used in trademarking. But again this would be part of the Brand schema, not of this identifier. 

```
{ 
  "@context" : "http://schema.org",
  "@type" : "Organization",
  "name" : "THE LINUX FOUNDATION",
  "brand": {
      "@type": "Brand",
      "name": "Kubernetes",
      "image": {
          "@type": "ImageObject",
          "url": "http://example.com/kubernetes.jpg",
          "description": "a triangular shaped chocolate",
        },
      "identifier": [{
          "@type": "Trademark"
          "registrationNumber": "4754865",
          "applicationNumber": "86400668",
          "trademarkOfficeCode": "USPTO",
          "class": "9"
        }, {
          "@type": "Trademark"
          "registrationNumber": "013794904",
          "trademarkOfficeCode": "EUTM",
          "class": "9"
        }]
  }

  [...]

}
```

**Comment by fili** [comment]
@thadguidry This sounds like a good compromise:

> The use of the term Registration Number is used in multiple systems of the USPTO (yes, very confusing, I know) ... its best to use a full name like "markRegistrationID" for the registered "Mark" ID to avoid confusion, or just simply "markID" and a good definition to go into Schema.org

Keep in mind that we only need to assist with the identification of the trademark. We don't need to validate the value of the identifier if the same series of numbers are also used for patents or attorneys, as long as the number used in the marked up Brand is identifiable and unique to this particular trademark. In other words, can I use the provided number to look up just this one trademark registration. If the answer is yes, then the requirements have been met.

I also adapted the other fields to match your suggestion for the registration number. Because of the difference between application number and registration number, I favour the longer versions for clarity sake.

In addition, I consider the following two fields as required fields for any lookup:

- markRegistrationID (Text or array of Text)
- markOfficeCode (Text or array of Text)

The following field is recommended as it can greatly improve accuracy and give some additional metadata which can be used for programmatic identification purposes (e.g. trademarks can be registered in different classes and the class system exists for a reason, as such this can assist in highlighting the difference between multiple different trademark registrations on the same brand name):

- markClass (Text or array of Text)

The last field here is mainly here for the US system where the application number tends to be different from the actual registration number, as such this is a nice to have optional field but not required.

- markApplicationID (Text or array of Text)

This now looks like:

```
{ 
  "@context" : "http://schema.org",
  "@type" : "Organization",
  "name" : "THE LINUX FOUNDATION",
  "brand": {
      "@type": "Brand",
      "name": "Kubernetes",
      "identifier": [{
          "@type": "Trademark"
          "markRegistrationID": "4754865",
          "markApplicationID": "86400668",
          "markOfficeCode": "USPTO",
          "markClass": "9"
        }, {
          "@type": "Trademark"
          "markRegistrationID": "013794904",
          "markOfficeCode": "EUTM",
          "markClass": "9"
        }]
  }

  [...]

}
```

As you I have dealt both directly and indirectly with registration of trademarks, brand and trade names, but only internationally in several countries and not in the US. So although you say you are not an expert I do appreciate your contribution to this :)

**Comment by MichaelAndrews-RM** [comment]
>Right now this proposal is about adding an type of identifier to a Brand. What the brand is, or how it is described, depends on the Brand schema.

That's a good point -- the trademark protects the brand.  The brand can be defined by many characteristics: name, logo, slogans, sound signatures, packaging shapes, unique typography, special colors, scents, and so on.  Right now the Brand only addresses name and logo (with slogan being a proposed addition.)  "Word marks" (which cover names and slogans) are [only one type of trademark.](https://www.prv.se/en/trademarks/applying-for-a-trademark/what-is-a-trademark/different-types-of-trademark/)  

One company or brand may have more than one trademark registration, for different aspects of brand.  The name changes, though the logo doesn't (or vice versa).  Or they register a trademarked packaging shape to their brand.  I think the trademark-related properties will need to indicate the [kind of trademark:](https://www.ige.ch/en/protecting-your-ip/trade-marks/before-you-apply/your-ip-protection-strategy/trade-mark-types.html)

* Word mark
* Figurative mark
* Combined word/figurative mark
* Three-dimensional trade mark
* Acoustic mark

**Comment by fili** [comment]
@MichaelAndrews-RM 

> I think the trademark-related properties will need to indicate the kind of trademark

We can add the following recommended field (as I shortly referred to in the original post as trademarkType), in naming consistent with the other fields:

- markType (Text or array of Text)

This now looks like:

```
{ 
  "@context" : "http://schema.org",
  "@type" : "Organization",
  "name" : "THE LINUX FOUNDATION",
  "brand": {
      "@type": "Brand",
      "name": "Kubernetes",
      "identifier": [{
          "@type": "Trademark"
          "markRegistrationID": "4754865",
          "markApplicationID": "86400668",
          "markOfficeCode": "USPTO",
          "markClass": "9",
          "markType": "word"
        }, {
          "@type": "Trademark"
          "markRegistrationID": "013794904",
          "markOfficeCode": "EUTM",
          "markClass": "9"
          "markType": "word"
        }]
  }

  [...]

}
```

However I still feel that this may need to be part of the Brand schema instead. The proposed example above already supports multiple trademark registrations for identifier, no matter which type of trademark. As you mentioned, a brand can have multiple trademarks associated with it. If we move the details of the type of mark into the proposed trademark identifier, then does the proposed slogan addition on the Brand schema also need to be moved?

**Comment by thadguidry** [comment]
@fili Yes, your last example is well-fitting semantically for USPTO applications and the Madrid Protocol.

I also feel like @RichardWallis does have a good starting point in using PropertyValue for now.  Nothing stops you however from progressing on this proposal, and I think your last example brings it to a good state of capturing Trademark identifiers.

**Comment by danbri** [comment]
Can we get by here with the identifier + PropertyValue construction, at least until someone steps up and says they're expecting to consume/use the markup.

**Comment by nicolastorzec** [comment]
+1 for Richard's suggestion and keeping it simple until there's a strong use case.

**Comment by fili** [comment]
I don't think the PropertyValue will solve, so I will check in the coming days if I can add a commit to this describing the scenario described above.

**Comment by danbri** [comment]
No need to make commits / pull requests until we've determined that we want to do this.

This might also be something to consider for Wikidata.org instead...?

**Comment by thadguidry** [comment]
@fili
- You can participate in the Wikidata Patents discussion by starting here: https://www.wikidata.org/wiki/Wikidata:WikiProject_Patents , specifically the Trademarks sub-discussion https://www.wikidata.org/w/index.php?title=Wikidata_talk:WikiProject_Patents&oldid=710355461
- And also joining the Wikidata mailing list to ask questions here: https://lists.wikimedia.org/mailman/listinfo/wikidata
